### ISSUE TITLE
Isolate.resolvePackageUri throws for an unknown package

### ISSUE THREAD

**Issue by nex3** [dart-lang/sdk#25956]
The code `Isolate.resolvePackageUri(Uri.parse("package:asdf/asdf.dart"))` throws the following exception:

```
No mapping for 'asdf' package when resolving 'package:asdf/asdf.dart'.
#0      _resolvePackageUri (dart:_builtin:296)
#1      _resolvePackageUriFuture.<_resolvePackageUriFuture_async_body> (dart:_builtin:757)
#2      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
#3      _microtaskLoop (dart:async/schedule_microtask.dart:41)
#4      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#5      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)
```

The [method's documentation](https://api.dartlang.org/1.14.2/dart-isolate/Isolate/resolvePackageUri.html) says:

> If there is no valid mapping from the package: URI in the current isolate, then this call returns `null`.

Note that the exception is also a String, which core libraries should never throw.

**Comment by lrhn** [comment]
The error comes from the VM builtin library. 
I guess the dart:isolate code that calls it will need to catch the error and return null.

**Comment by kasperl** [comment]
Can you take a look at this, @iposva-google?

**Comment by iposva-google** [comment]
Out for review: https://codereview.chromium.org/1822863002

**Comment by iposva-google** [comment]
Fixed with 3b2a695d71057f3a538f79f48a96f6bdca2474bb
